From 7a1c680a2b140e2bab7d1b8a85b3dc97c8a131ba Mon Sep 17 00:00:00 2001
From: Apollon Oikonomopoulos <apollon@skroutz.gr>
Date: Mon, 19 Mar 2018 14:21:22 +0200
Subject: [PATCH] rapidjson 1.1 compatibility

RapidJSON 1.1 slightly changed its API for custom handlers, by
introducing the RawNumber method, see [1]. Instead of defining the
method itself, change all handlers to inherit from
rapidjson::BaseReaderHandler.

Additionally, the facter::facts::<anon>::stream_adapter must typedef Ch.

[1] https://github.com/Tencent/rapidjson/issues/581
---
 lib/inc/facter/facts/collection.hpp     | 1 +
 lib/src/facts/external/json_resolver.cc | 2 +-
 lib/src/facts/resolvers/gce_resolver.cc | 2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/inc/facter/facts/collection.hpp b/lib/inc/facter/facts/collection.hpp
index c89daa5e..570cc67d 100644
--- a/lib/inc/facter/facts/collection.hpp
+++ b/lib/inc/facter/facts/collection.hpp
@@ -46,6 +46,7 @@ namespace facter { namespace facts {
          */
         struct stream_adapter
         {
+            typedef char Ch;
             /**
              * Constructs an adapter for use with rapidjson around the given stream.
              * @param stream an output stream to which JSON will be written
diff --git a/lib/src/facts/external/json_resolver.cc b/lib/src/facts/external/json_resolver.cc
index 8edc0467..35817d59 100644
--- a/lib/src/facts/external/json_resolver.cc
+++ b/lib/src/facts/external/json_resolver.cc
@@ -24,7 +24,7 @@ using namespace rapidjson;
 namespace facter { namespace facts { namespace external {
 
     // Helper event handler for parsing JSON data
-    struct json_event_handler
+    struct json_event_handler : public rapidjson::BaseReaderHandler<>
     {
         explicit json_event_handler(collection& facts) :
             _initialized(false),
diff --git a/lib/src/facts/resolvers/gce_resolver.cc b/lib/src/facts/resolvers/gce_resolver.cc
index ed579e5f..6ea737fa 100644
--- a/lib/src/facts/resolvers/gce_resolver.cc
+++ b/lib/src/facts/resolvers/gce_resolver.cc
@@ -35,7 +35,7 @@ namespace facter { namespace facts { namespace resolvers {
 #endif
 
     // Helper event handler for parsing JSON data
-    struct gce_event_handler
+    struct gce_event_handler : public rapidjson::BaseReaderHandler<>
     {
         explicit gce_event_handler(map_value& root) :
             _initialized(false),
-- 
2.16.1

